tests/test_unit/test_logging.py

```diff
@@ -45,27 +46,42 @@ def test_log_warning(caplog):
     assert caplog.records[0].levelname == "WARNING"
 
 
-@pytest.mark.parametrize("input_data", ["dataset", "dataarray"])
-def test_log_to_attrs(input_data, valid_poses_dataset):
+@pytest.mark.parametrize(
+    "input_data",
+    [
+        "valid_poses_dataset",
+        "valid_bboxes_dataset",
+    ],
+)
+@pytest.mark.parametrize(
+    "selector_fn, expected_selector_type",
+    [
+        (lambda ds: ds, xr.Dataset),  # take full dataset
+        (lambda ds: ds.position, xr.DataArray),  # take position data array
+    ],
+)
+def test_log_to_attrs(
+    input_data, selector_fn, expected_selector_type, request
+):
     """Test that the ``log_to_attrs()`` decorator appends
-    log entries to the output data's ``log`` attribute and
-    checks that ``attrs`` contains all expected values.
+    log entries to the dataset's or the data array's ``log``
+    attribute and check that ``attrs`` contains all the expected values.
     """
 
+    # a fake operation on the dataset to log
     @log_to_attrs
     def fake_func(data, arg, kwarg=None):
         return data
 
-    input_data = (
-        valid_poses_dataset
-        if input_data == "dataset"
-        else valid_poses_dataset.position
-    )
+    # apply operation to dataset or data array
+    dataset = request.getfixturevalue(input_data)
+    input_data = selector_fn(dataset)
     output_data = fake_func(input_data, "test1", kwarg="test2")
 
+    # check the log in the dataset is as expected
+    assert isinstance(output_data, expected_selector_type)
     assert "log" in output_data.attrs
     assert output_data.attrs["log"][0]["operation"] == "fake_func"
-    assert (
-        output_data.attrs["log"][0]["arg_1"] == "test1"
-        and output_data.attrs["log"][0]["kwarg"] == "test2"
-    )
+    assert output_data.attrs["log"][0]["arg_1"] == "test1"
+    assert output_data.attrs["log"][0]["kwarg"] == "test2"
```

Comment:
the exact same line is repeated twice?

Reply:
line 86 and 87

Reply:
yes, thanks for spotting it! 

Sometimes vscode does this with the last line in a file, I _think_ when I ctrl+s to force save a file. I still have no idea why, but there is at least [one report ](with the last line in the file)of a similar bug